malloc/src/mymalloc.c

```diff
@@ -4,19 +4,80 @@
 
 #define ALIGNMENT 8 // sizeof(double)
 #define ALIGN(size) (((size) + (ALIGNMENT - 1)) & ~(ALIGNMENT))
-#define HEADER_SIZE sizeof(header_t)
+#define HEADER_SIZE sizeof(header_t)/sizeof(double) + (sizeof(header_t)%sizeof(double) != 0) //size of header in blocks of 8 bytes
+#define HEAP_SIZE 512
+#define END &heap[HEAP_SIZE]  //one past the end of the heap
+
+static double heap[HEAP_SIZE+1]; 
 
 typedef struct memory_header {
-    int size;
+    int size; 
     bool isFree;
 } header_t;
 
+//given a block header, return the next block, or ptr to one past heap if it does not exist
+header_t* nextHeader(header_t *header){
+    return (header_t *) &header[header->size + HEADER_SIZE];
```

Comment:
it looks like `header->size` is in terms of doubles, and this is being casted to a `(header_t *)` ptr. this ends up working only because `sizeof(double) == sizeof(header_t)`, but wouldn't work otherwise I think.

Reply:
`return (header_t *) ((double *) header + header->size) + HEADER_SIZE;` will work better, i think